src/TimingData.cpp

```diff
@@ -862,13 +859,10 @@ void TimingData::GetBeatInternal(GetBeatStarts& start, GetBeatArgs& args,
 						args.bps_out= bps;
 						return;
 					}
-					start.last_time= next_event_time;
-					INC_INDEX(start.delay);
-					if(event_type == FOUND_DELAY)
-					{
-						break;
-					}
-					[[fallthrough]];
```

Comment:
??? don't just remove the fallthrough because you don't know what it's doing.

Fallthrough is a c++17 keyword to deliberated indicate you want the fallthrough to the next case. If the event_type is not a delay we want to fall through to the stop.

If you're removing macros make sure that those PRs are *all* that you're doing. Any code changes should come afterwards.

Reply:
> ??? don't just remove the fallthrough because you don't know what it's doing.
> 
> Fallthrough is a c++17 keyword to deliberated indicate you want the fallthrough to the next case. If the event_type is not a delay we want to fall through to the stop.
> 
> If you're removing macros make sure that those PRs are _all_ that you're doing. Any code changes should come afterwards.

I'm awfully sorry, I had already realized this but apparently I never pushed it 🤦‍♀️

I meant to push

```
			case FOUND_STOP_DELAY:
				{
					const DelaySegment* ss= ToDelay(delays[start.delay]);
					time_to_next_event= ss->GetPause();
					next_event_time= start.last_time + time_to_next_event;
					if(args.elapsed_time < next_event_time)
					{
						args.freeze_out= false;
						args.delay_out= true;
						args.beat= ss->GetBeat();
						args.bps_out= bps;
						return;
					}
					start.last_time = next_event_time;
					++start.delay;
					++curr_segment;
					[[fallthrough]]; // FOUND_STOP_DELAY should fall thru here.
				}
			case FOUND_STOP:
				{
					const StopSegment* ss= ToStop(stops[start.stop]);
					time_to_next_event= ss->GetPause();
					next_event_time= start.last_time + time_to_next_event;
					if(args.elapsed_time < next_event_time)
					{
						args.freeze_out= true;
						args.delay_out= false;
						args.beat= ss->GetBeat();
						args.bps_out= bps;
						return;
					}
					start.last_time= next_event_time;
					++start.stop;
					++curr_segment;
					break; // FOUND_STOP should break here
				}
```

But for now I'll limit this to only removing the macro.